src/H5Gtraverse.c

```diff
@@ -665,8 +667,15 @@ H5G__traverse_real(const H5G_loc_t *_loc, const char *name, unsigned target, H5G
                     pline = &def_pline;
 
                 /* Create the intermediate group */
-                /* XXX: Should we allow user to control the group creation params here? -QAK */
-                gcrt_info.gcpl_id    = H5P_GROUP_CREATE_DEFAULT;
+                gcrt_info.gcpl_id = H5P_GROUP_CREATE_DEFAULT;
+                /* Propagate the object creation properties when creating intermedidate groups */
+                if ((target & H5G_CRT_OBJ) && (ocrt_info = H5L_OCRT_INFO(op_data)) != NULL) {
+                    if (ocrt_info->obj_type == H5O_TYPE_GROUP)
+                        gcrt_info.gcpl_id = H5G_OBJ_ID(ocrt_info->crt_info);
+                    else if (ocrt_info->obj_type == H5O_TYPE_DATASET)
+                        gcrt_info.gcpl_id = H5D_OBJ_ID(ocrt_info->crt_info);
```

Comment:
Seems like this may be problematic to assign a DCPL ID to a GCPL ID. They're both subclasses of H5P_CLS_OBJECT_CREATE_g, but if the code ever needs to retrieve some group-specific properties (maybe in the future), it seems like that may fail. Should `gcrt_info` have an `ocpl_id` field rather than a `gcpl_id` field to try to make it clear that only generic OCPL properties should be retrieved from it?

Reply:
> Seems like this may be problematic to assign a DCPL ID to a GCPL ID. They're both subclasses of H5P_CLS_OBJECT_CREATE_g, but if the code ever needs to retrieve some group-specific properties (maybe in the future), it seems like that may fail. Should `gcrt_info` have an `ocpl_id` field rather than a `gcpl_id` field to try to make it clear that only generic OCPL properties should be retrieved from it?

Agree

Reply:
I have couple questions about changing gcpl_id to ocpl_id:
--Changing gcpl_id field in gcrt_info to ocpl_id would lead to changes in quite a few files which uses gcrt_info.gpcl_id.
 --If I change gcpl_id in gcrt_info which is H5G_obj_create_t struct, should I also change dcpl_id field in H5D_obj_create_t struct to ocpl_id?

Reply:
> --Changing gcpl_id field in gcrt_info to ocpl_id would lead to changes in quite a few files which uses gcrt_info.gpcl_id.

It's probably fine to put this off to a separate focused PR since the code appears to currently only retrieve generic properties.

> --If I change gcpl_id in gcrt_info which is H5G_obj_create_t struct, should I also change dcpl_id field in H5D_obj_create_t struct to ocpl_id?

As far as I can tell, it looks like the only place that sets up a `H5D_obj_create_t` struct sets its `dcpl_id` field to a DCPL, so that one shouldn't need to be changed.